rosetta/docs/NATIVE_FP8.md

```diff
@@ -128,7 +128,61 @@ python -m paxml.main \
 
 Please ensure you include the first two flags, `--xla_gpu_enable_reduction_epilogue_fusion=false` and `--xla_gpu_enable_triton_gemm=false`, as they are essential for enabling the FP8 functionality. The additional flags primarily focus on performance enhancement and should also prove beneficial for non-FP8 executions.
 
-### Transformer Engine vs Native FP8 Support
+## Guide for Ninja Users
+
+### Exact pattern that XLA can match for FP8 MatMul
+The specific graph pattern that XLA supports for FP8 matmul is illustrated below:
+
+```
+  convert -> multiply -> (x) -> dot
+broadcast ->
+
+# or
+
+  convert -> (x) -> dot
+```
+
+XLA will pattern match the above and rewrite it to FP8 matmul when:
+1. `convert`: converts `f8` inputs to [`bf16`|`f16`|`f32`].
+2. `broadcast`: broadcasts a [`bf16`|`f16`|`f32`] scalar. The scalar will be used as the scaling factor of the inputs. Note, the `convert` and `broadcast` need to have the same output dtype. If `broadcast` (and `multiply`) is not provided, the scaling factor will be set to `1.`.
+3. `(x)`: an arbitrary number of these allowed ops:
+
+```
+Bitcast, Broadcast, Copy, DynamicSlice, Pad, Reshape, Select, Slice, Transpose,
+AllGather, AllToAll, CollectivePermute
+```
+
+### Gradient accumulation of FP8 params
+FP8 params, also known as `OverwriteWithGrad` params (or FP8 meta), may be shared across different iterations of a loop in the context of pipeline parallelism. During backpropagation, the autograd system accumulates their gradients from each iteration through the default addition operation. This is undesirable as addition is meaningless for FP8 params.
+
+To address this, we introduce a custom dtype wrapper `fp32_max_grad`. It tells the autograd system to perform the max operation for gradient accumulation. This aligns with our expectations for FP8 params. The basic usage is demonstrated below:
+
+```python
+from flax.linen import fp8_ops
+# ah and sf are short for amax history and scaling factor used as the FP8 params.
+f32 = jnp.float32
+fm32 = fp8_ops.fp32_max_grad
+def outer(x, ah_f32, sf_f32):
```

Comment:
Longer name than ah and sf? Or add a comment that tell what they mean?

Reply:
Done.